src/cirq_iqm/adonis.py

```diff
@@ -69,45 +72,42 @@ class Adonis(idev.IQMDevice):
     def operation_decomposer(self, op: cirq.Operation) -> Optional[list[cirq.Operation]]:
         """Decomposes gates into the native Adonis gate set.
         """
-        # NOTE: All the decompositions below keep track of global phase (required for decomposing
-        # controlled gates), but for now assume that op.gate.global_shift is zero.
-        # It seems that Cirq native decompositions ignore global phase entirely.
+        # All the decompositions below keep track of global phase (required for decomposing controlled gates).
+        # It seems that Cirq native decompositions ignore global phase entirely?
 
-        if isinstance(op.gate, ops.ISwapPowGate):
-            # the ISwap family is implemented using the XY interaction
-            s = -0.5 * op.gate.exponent
-            return [
-                ig.XYGate(exponent=s).on(*op.qubits)
-            ]
         if isinstance(op.gate, ops.CZPowGate):
-            # decompose CZPowGate using IsingGate
-            s = -0.5 * op.gate.exponent
-            L = ops.ZPowGate(exponent=-s, global_shift=-0.5)
+            # decompose CZPowGate using ZZPowGate
```

Comment:
Minor: This decomposition is shared between Adonis and Valkmusa, would it make sense to create separate shared functions for the decompositions used in both?

Reply:
That's currently the only shared part. Maybe later if there will be more of them.